wordpress-mobile/WordPress-iOS
# Comments: fix moderation options on Atomic sites

Fixes: #16859
Ref: #16783

This fixes two issues with Comment moderation on Atomic sites:
- Author/Contributor no longer have moderation options. They can now only Like and Reply to Approved comments. (This matches the intended behavior on #16783 )
- If replying to a comment on a closed post, the API's error message is displayed, which in this case is `Comments on this post are closed`.

To test:

---
On an Atomic site where you cannot moderate comments (your role is Author or Contributor):
- Go to My Site > Comments.
- Verify you cannot Approve, Spam, or Trash any comment type.
- Verify you can Like and Reply to Approved comments.
- Verify you cannot Like or Reply to any other comment type (Pending, Spam, Trashed).
  - Since no actions are available, the action bar is hidden.
  - Since you cannot Reply, the Reply bar at the bottom is hidden.


| Approved | Pending |
|--------|-------|
| ![approved](https://user-images.githubusercontent.com/1816888/124014594-f1d20600-d9a0-11eb-99ab-6da155ab4ae0.png) | ![pending](https://user-images.githubusercontent.com/1816888/124014607-f696ba00-d9a0-11eb-933d-6eedcb0f7c0d.png) |

---
On an Atomic site where you cannot moderate comments (your role is Author or Contributor):
- Go to My Site > Comments.
- Select an Approved comment on a post where commenting is closed.
- Reply to the comment.
- Verify the `Comments on this post are closed` message is displayed.

<kbd>![fail](https://user-images.githubusercontent.com/1816888/125703338-8d123741-7901-4f52-968b-2e9aa00f4752.png)</kbd>

---
On an Atomic site where you can moderate comments (your role is Editor or Administrator):
- Verify you can moderate, like, and reply to all comments.

<kbd>![god_mode](https://user-images.githubusercontent.com/1816888/124015014-72910200-d9a1-11eb-97eb-422be7a122ce.png)</kbd>

---
## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
